lib/auth/auth.go

```diff
@@ -5745,19 +5745,19 @@ func (a *Server) CreateAccessRequestV2(ctx context.Context, req types.AccessRequ
 
 	req.SetCreationTime(now)
 
-	// Always perform variable expansion on creation.
-	expandOpts := services.WithExpandVars(true)
-	if err := services.ValidateAccessRequestForUser(ctx, a.clock, a, req, identity, expandOpts); err != nil {
-		return nil, trace.Wrap(err)
-	}
-
 	// Look for user groups and associated applications to the request.
 	requestedResourceIDs, err := a.appendImplicitlyRequiredResources(ctx, req.GetRequestedResourceIDs())
 	if err != nil {
 		return nil, trace.Wrap(err, "adding additional implicitly required resources")
 	}
 	req.SetRequestedResourceIDs(requestedResourceIDs)
 
+	// Always perform variable expansion on creation.
+	expandOpts := services.WithExpandVars(true)
+	if err := services.ValidateAccessRequestForUser(ctx, a.clock, a, req, identity, expandOpts); err != nil {
+		return nil, trace.Wrap(err)
+	}
```

Comment:
`ValidateAccessRequestForUser` moved down because validation also involves role resolution, which is needed to resolve proper role for the implicitly added saml app.